pyvista/utilities/geometric_objects.py

```diff
@@ -161,14 +161,15 @@ def CylinderStructured(radius=0.5, height=1.0,
     grid.dimensions = [nr, theta_resolution+1, z_resolution]
 
     # Orient properly in user direction
-    direction = np.array(direction)
-    direction = direction / np.linalg.norm(direction)
-    vx = np.random.randn(3)
-    vx -= vx.dot(direction) * direction
-    vx /= np.linalg.norm(vx)
-    vy = np.cross(direction, vx)
-    rmtx = np.array([vx, vy, direction])
-    grid.points = grid.points.dot(rmtx)
+    vx = np.array([0., 0., 1.])
+    if not np.allclose(vx, direction):
+        direction = np.array(direction)
```

Comment:
What about `direction = np.array(direction, copy=False)` or  `direction = np.asarray(direction)` (the are basically the same).

Also, as you use `direction` anyway in the `if`, I would put the casting just one line before `if not np.allclose`.

Reply:
The `asarray` or `copy=False` avoids making a new copy if it is already an array.

Reply:
Is it even necessary at all to explicitly convert direction? Numpy methods can take "array-like" objects and return ndarrays.

Reply:
You are probably right. Might be one of those 'better safe than sorry'.

Reply:
Looking at the entire function makes me think that you are right, casting is not required at all as it is done by numpy itself (problems can happen without casting if you try to access numpy-attributes like `dtype`, `max`).

Reply:
@prisae, you point on 'better safe than sorry' is exactly why I did that - which was mostly just from habit. It shouldn't be needed for this particular function

```suggestion
```